### PR TITLE
Calico iptables backend config parameter

### PR DESCRIPTION
**Pull request by makhov**
Signed-off-by: Alexey Makhov <amakhov@mirantis.com>

## Description

A new config parameter is added to set which iptables backend Calico should use.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/apis/k0s.k0sproject.io/v1beta1/calico.go [hunk 1]
```diff
@@ -64,6 +67,7 @@ func DefaultCalico() *Calico {
 		Overlay:                 "Always",
 		IPAutodetectionMethod:   "",
 		IPv6AutodetectionMethod: "",
+		IPTablesBackend:         "auto",
```
Comment (jnummelin) [inline pkg/apis/k0s.k0sproject.io/v1beta1/calico.go:70]: Calico docs says the default is `[Default: Auto]`, note the capital `A`. dunno if it matter though but to be on safe side I'd change it to the exact same in k0s side. See https://projectcalico.docs.tigera.io/reference/felix/configuration#iptables-dataplane-configuration

Comment (makhov) [reply pkg/apis/k0s.k0sproject.io/v1beta1/calico.go]: Yeah, fixed. Thanks!

**Comment by jnummelin** [comment]
This does makes me wonder if we'd need to generic way for a user to pass in ANY set of env variables. there's quite a few envs one can configure with calico

**Comment by makhov** [comment]
Yeah, we definitely need it. The question is whether to do it in this PR or in a separate one.

My first idea was that since we set `FELIX_IPTABLESBACKEND` in the calico-node daemonset, we need to have a parameter to change the value of the var.
On the other hand, the default value is `Auto`, so we can just remove it and let the user decide to set `FELIX_IPTABLESBACKEND` or not.

**Comment by jnummelin** [comment]
> The question is whether to do it in this PR or in a separate one.

Maybe better to do it in separate PR. This PR might need to get backported to release branches.

**Comment by jnummelin** [comment]
Related https://github.com/projectcalico/calico/issues/5760

**Comment by makhov** [comment]
Closing, since it can be done by setting env var (needs #2269 to be merged)

**Pull request closed without merging**